File: UCSample/UCSample/NumpadViewController.m
//
//  SecondViewController.m
//  UCSample
//

//  Copyright (c) 2013 PortSIP Solutions, Inc. All rights reserved.
//

#import "NumpadViewController.h"
#import "LineTableViewController.h"
#import "AppDelegate.h"

#define kTAGStar 10
#define kTAGSharp 11

#define kTAGVideoCall 12
#define kTAGAudioCall 13
#define kTAGHangUp 14

#define kTAGHold 15
#define kTAGUnHold 16
#define kTAGRefer 17

#define kTAGMute 18
#define kTAGSpeak 19
#define kTAGStatistics 20

#define kTAGDelete 21

@interface NumpadViewController ()

@end

@implementation NumpadViewController

- (void)viewDidLoad {
  [super viewDidLoad];

  _textNumber.delegate = self;

  [_labelStatus setText:@""];
  // Do any additional setup after loading the view, typically from a nib.
}

- (void)viewWillAppear:(BOOL)animated {
  AppDelegate *appDelegate =
      (AppDelegate *)[[UIApplication sharedApplication] delegate];

  [_buttonLine
      setTitle:[NSString stringWithFormat:@"Line%zd:", appDelegate.activeLine]
      forState:UIControlStateNormal];
  [super viewWillAppear:YES];
}

- (void)didReceiveMemoryWarning {
  [super didReceiveMemoryWarning];
  // Dispose of any resources that can be recreated.
}

- (BOOL)textFieldShouldReturn:(UITextField *)textField {
  [textField resignFirstResponder];
  return YES;
}

- (IBAction)onButtonClick:(id)sender {
  NSInteger tag = ((UIButton *)sender).tag;
  AppDelegate *appDelegate =
      (AppDelegate *)[[UIApplication sharedApplication] delegate];

  switch (tag) {
  case 0:
  case 1:
  case 2:
  case 3:
  case 4:
  case 5:
  case 6:
  case 7:
  case 8:
  case 9: {
    _textNumber.text = [_textNumber.text
        stringByAppendingString:[NSString stringWithFormat:@"%zd", (long)tag]];

    char dtmf = tag;
    [appDelegate pressNumpadButton:dtmf];
  } break;
  case kTAGStar: {
    _textNumber.text = [_textNumber.text stringByAppendingString:@"*"];
    [appDelegate pressNumpadButton:10];
  } break;

  case kTAGSharp: {
    _textNumber.text = [_textNumber.text stringByAppendingString:@"#"];
    [appDelegate pressNumpadButton:11];
  } break;
  case kTAGDelete: {
    NSString *number = _textNumber.text;
    if ([number length] > 0) {
      _textNumber.text = [number substringToIndex:([number length] - 1)];
    }
    break;
  }
  case kTAGVideoCall: {
    [appDelegate makeCall:[_textNumber text] videoCall:YES];
    break;
  }
  case kTAGAudioCall: {
    [appDelegate makeCall:[_textNumber text] videoCall:NO];
    break;
  }

  case kTAGHangUp: {
    [appDelegate hungUpCall];
    break;
  }
  case kTAGHold: {
    [appDelegate holdCall];
    break;
  }
  case kTAGUnHold: {
    [appDelegate unholdCall];
    break;
  }

  case kTAGRefer: {
    [appDelegate referCall:[_textNumber text]];
  } break;
  case kTAGMute: {
    UIButton *buttonMute = (UIButton *)sender;
    if ([[[buttonMute titleLabel] text] isEqualToString:@"unMute"]) {
      [appDelegate muteCall:NO];

      [buttonMute setTitle:@"Mute" forState:UIControlStateNormal];
      [_labelStatus setText:@"Mute"];
    } else {
      [appDelegate muteCall:YES];

      [buttonMute setTitle:@"unMute" forState:UIControlStateNormal];
      [_labelStatus setText:@"unMute"];
    }
    break;
  }
  case kTAGSpeak: {
    UIButton *buttonSpeaker = (UIButton *)sender;
    if ([[[buttonSpeaker titleLabel] text] isEqualToString:@"Speaker"]) {
      [appDelegate setLoudspeakerStatus:YES];

      [buttonSpeaker setTitle:@"earphone" forState:UIControlStateNormal];
      [_labelStatus setText:@"Enable Speaker"];
    } else {
      [appDelegate setLoudspeakerStatus:NO];

      [buttonSpeaker setTitle:@"Speaker" forState:UIControlStateNormal];
      [_labelStatus setText:@"Disable Speaker"];
    }
  } break;
  case kTAGStatistics: {
    [appDelegate makeTest];
    //[appDelegate getStatistics];
  } break;
  }
}

- (IBAction)onLineClick:(id)sender {
  AppDelegate *appDelegate =
      (AppDelegate *)[[UIApplication sharedApplication] delegate];
  [appDelegate switchSessionLine];
}

- (void)setStatusText:(NSString *)statusText {
  [_labelStatus setText:statusText];
  NSLog(@"%@", statusText);
}

@end

File: UCSample_swift/UCSample/NumpadViewController.swift
//
//  SecondViewController.m
//  UCSample
//

//  Copyright (c) 2013 PortSIP Solutions, Inc. All rights reserved.
//

import UIKit

class NumpadViewController: UIViewController, UITextFieldDelegate {
    let kTAGStar: Int32 = 10
    let kTAGSharp: Int32 = 11

    let kTAGVideoCall: Int32 = 12
    let kTAGAudioCall: Int32 = 13
    let kTAGHangUp: Int32 = 14

    let kTAGHold: Int32 = 15
    let kTAGUnHold: Int32 = 16
    let kTAGRefer: Int32 = 17

    let kTAGMute: Int32 = 18
    let kTAGSpeak: Int32 = 19
    let kTAGStatistics: Int32 = 20

    let kTAGDelete: Int32 = 21

    @IBOutlet var textNumber: UITextField!
    @IBOutlet var labelStatus: UILabel!
    @IBOutlet var buttonLine: UIButton!

    var status: String = ""

    override func viewDidLoad() {
        super.viewDidLoad()

        textNumber.delegate = self

        labelStatus.text = status
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func viewWillAppear(_: Bool) {
        let appDelegate = UIApplication.shared.delegate as! AppDelegate

        buttonLine.setTitle("Line: \(appDelegate._activeLine!)", for: .normal)
        super.viewWillAppear(true)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
        textField.resignFirstResponder()
        return true
    }

    @IBAction func onButtonClick(_ sender: AnyObject) {
        let tag = Int32((sender as! UIButton).tag)
        let appDelegate = UIApplication.shared.delegate as! AppDelegate
        switch tag {
        case 0, 1, 2, 3, 4, 5, 6, 7, 8, 9:

            textNumber.text = textNumber.text! + String(tag)
            let dtmf = tag
            appDelegate.pressNumpadButton(dtmf)

        case kTAGStar:
            textNumber.text = (textNumber.text)! + "*"
            appDelegate.pressNumpadButton(10)

        case kTAGSharp:
            textNumber.text = (textNumber.text)! + "#"
            appDelegate.pressNumpadButton(11)

        case kTAGDelete:

            if !textNumber.text!.isEmpty {
                var text = textNumber.text!
                text.remove(at: text.index(before: text.endIndex))
                textNumber.text = text
            }

        case kTAGVideoCall:
            _ = appDelegate.makeCall(textNumber.text!, videoCall: true)

        case kTAGAudioCall:
            _ = appDelegate.makeCall(textNumber.text!, videoCall: false)

        case kTAGHangUp:
            appDelegate.hungUpCall()

        case kTAGHold:
            appDelegate.holdCall()

        case kTAGUnHold:
            appDelegate.unholdCall()

        case kTAGRefer:
            appDelegate.referCall(textNumber.text!)

        case kTAGMute:
            let buttonMute = sender as! UIButton
            if buttonMute.titleLabel?.text == "unMute" {
                appDelegate.muteCall(false)

                buttonMute.setTitle("Mute", for: .normal)
                labelStatus.text = "Mute"
            } else {
                appDelegate.muteCall(true)

                buttonMute.setTitle("unMute", for: .normal)
                labelStatus.text = "unMute"
            }

        case kTAGSpeak:

            let buttonSpeaker = sender as! UIButton
            if buttonSpeaker.titleLabel?.text == "Speaker" {
                appDelegate.setLoudspeakerStatus(true)

                buttonSpeaker.setTitle("earphone", for: .normal)
                labelStatus.text = "Enable Speaker"
            } else {
                appDelegate.setLoudspeakerStatus(false)

                buttonSpeaker.setTitle("Speaker", for: .normal)
                labelStatus.text = "Disable Speaker"
            }
        case kTAGStatistics:
            appDelegate.getStatistics()
        default: break
        }
    }

    @IBAction func onLineClick(_: AnyObject) {
        let appDelegate = UIApplication.shared.delegate as! AppDelegate
        appDelegate.switchSessionLine()
    }

    func setStatusText(_ statusText: String) {
        status = statusText
        if labelStatus != nil {
            labelStatus.text = statusText
        }
        print(statusText)
    }
}
